File: 26.cs
public class Solution {
    public int RemoveDuplicates(int[] nums) {
        int removed = 0;
        int i = 1;
        int n = nums.Length;
        while (i < n) {
            if (nums[i] == nums[i-1]) {
                int j = i;
                while (j < n - 1) {
                    nums[j] = nums[j+1];
                    j++;
                }
                n--;
                i--;
            }
            i++;
        }
        return n;

    }
}

File: code/26.cpp
class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        int n = nums.size();
        if (n == 1) {
            return 1;
        }
        int i = 0;
        int j = 0;
        int k = 1;
        while (j < n - 1) {
            j++;
            if (nums[j] != nums[j-1]) {
                // We've found a new element
                i++;
                nums[i] = nums[j];
                k++;
            }
        }
        return k;        
    }
};
